secondarylib-feedback/src/main/scala/com/socrata/datacoordinator/secondary/feedback/FeedbackSecondary.scala

```diff
@@ -362,9 +349,13 @@ abstract class FeedbackSecondary[CT,CV] extends Secondary[CT,CV] {
       // update cookie
       val copyNumber = CopyNumber(copyInfo.copyNumber)
 
-      val primaryKey = schema.filter {
-        case (colId, colInfo) => colInfo.isUserPrimaryKey
-      }.toSeq.map({ case (_, colInfo) => colInfo.id}).headOption.getOrElse(systemId)
+      val systemId = schema.filter {
+        case (_, colInfo) => colInfo.isSystemPrimaryKey
+      }.toSeq.map({ case (_, colInfo) => colInfo.id}).headOption.getOrElse {
+        // this shouldn't ever happen
+        log.error("Resync done on dataset with no system id column!!! Schema: {}", schema)
+        throw new Exception("Resync done on dataset with no system id column!")
```

Comment:
Would this make more sense as a `BrokenDatasetSecondaryException`?

Reply:
That would make sense